### PR TITLE
Cloud CLI integration

### PR DESCRIPTION
**Pull request by jackkleeman**
- New config file mechanism inc `restate config` subcommand
- `restate cloud login`
- `restate cloud environment configure`

### CODE REVIEW

File: cli/src/clients/cloud/client.rs [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+}
+
+/// A lazy wrapper around a reqwest response that deserializes the body on
+/// demand and decodes our custom error body on non-2xx responses.
+pub struct Envelope<T> {
+    inner: reqwest::Response,
+
+    _phantom: std::marker::PhantomData<T>,
+}
+
```
Comment (AhmedSoliman) [inline cli/src/clients/cloud/client.rs:42]: Is it intentional to copy the code from metas_client or is it possible to reuse existing implementation?

File: cli/src/commands/cloud/environments/configure.rs [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+                    "Couldn't find environment {environment} in account {}",
+                    accounts[account_i].name,
+                ))
+            }
+        }
+    } else {
+        match environments.len() {
+            0 => {
+                return Err(anyhow::anyhow!(
+                "No environments set up; use the Restate Cloud UI to create your first environment"
```
Comment (AhmedSoliman) [inline cli/src/commands/cloud/environments/configure.rs:132]: Do you think we should return this as an anyhow error, or should we take control over the formatting and eprintln instead?

**Review by AhmedSoliman — APPROVED** [review]
Took a quick look, no issues from my side. +1 to merge.

File: cli/Cargo.toml [hunk 3]
```diff
@@ -44,20 +47,27 @@ is-terminal = { version = "0.4.9" }
...
+toml_edit = "0.22.12"
 tracing = { workspace = true }
 tracing-log = { version = "0.2.0" }
 tracing-subscriber = { workspace = true }
 unicode-width = { version = "0.1.11" }
 url = { version = "2.4.1" }
+uuid = { workspace = true }
 zip = "0.6"
 zip-extensions = "0.6"
+typify = "0.1.0"
```
[the remark this answers is not in the document]
Comment (jackkleeman) [reply cli/Cargo.toml:70]: sorry i think the multiline thing is my editor zed - will see if i can configure that

**Pull request merged**